packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -818,5 +818,29 @@ describe('ProjectDataSource', () => {
 
       expect(defaultSpecFileName).to.equal('cypress/component-tests/foo/ComponentName.spec.js')
     })
+
+    describe('jsx/tsx handling', () => {
```

Comment:
Could we have coverage for each possibility now? `.js`, `.ts`, `.jsx`, `.tsx`? Is this unwieldy to test?

Reply:
Got a bit tricker when I looked at stubbing out the TS case here. Since all this logic does is take the "js | ts" default extension and add "x" I didn't consider it necessary to cover all four permutations, but happy to look into it if you think it's of value

Reply:
Yeah it looks like `fileExtensionToUse` itself is asserted implicitly plenty, I think this is fine.